src/tools/utils/editorOperationUtils.ts

```diff
@@ -0,0 +1,114 @@
+import * as monaco from 'monaco-editor'
+import { ValidAnnotatedEditOperation } from 'vscode/vscode/vs/editor/common/model'
+import { getLockedRanges, minusRanges } from './rangeUtils'
+
+function createNewOperation (
+  oldOperation: ValidAnnotatedEditOperation,
+  newRange: monaco.Range,
+  newText: string | null,
+  index: number
+): ValidAnnotatedEditOperation {
+  const identifier = oldOperation.identifier != null
+    ? { major: oldOperation.identifier.major, minor: oldOperation.identifier.minor + index }
+    : null
+  return new ValidAnnotatedEditOperation(
+    identifier,
+    newRange,
+    newText,
+    oldOperation.forceMoveMarkers,
+    oldOperation.isAutoWhitespaceEdit,
+    oldOperation._isTracked
+  )
+}
+
+function createNewOperationsFromRanges (
+  oldOperation: ValidAnnotatedEditOperation,
+  editableRanges: monaco.Range[],
+  splitText: (string | null)[]
+): ValidAnnotatedEditOperation[] {
+  if (editableRanges.length <= 0 || splitText.length <= 0) {
+    return [oldOperation]
+  }
+
+  return splitText.map((text, index) => createNewOperation(oldOperation, editableRanges[index]!, text, index))
+}
+
+function splitOperationText (
```

Comment:
What about `tryIgnoreLockedCodeForOperation`?

Reply:
[refactor: rename functions on locked code and their variables](https://github.com/CodinGame/monaco-editor-wrapper/pull/125/commits/4211c36f490bd0ebf871eeb479fac4afd597cb2d)